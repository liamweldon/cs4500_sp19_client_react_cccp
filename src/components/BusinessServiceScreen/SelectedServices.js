import React from "react";

const SelectedServices = props => {
  const services = props.services;
  const selectedServices = props.selectedServices;

  return (
    <div>
      <h4>Selected Services</h4>
      <br />
    </div>
  );
};

export default SelectedServices;
