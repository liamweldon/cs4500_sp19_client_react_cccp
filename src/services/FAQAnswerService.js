import {API_ROOT} from '../api-config';

export default class FAQAnswerService {
    static instance = null;
    static getInstance() {
        if (FAQAnswerService.instance === null) {
            FAQAnswerService.instance = new FAQAnswerService();
        }
        return this.instance;
    }
    findFAAById = (id) => fetch(`${API_ROOT}/api/faas/${id}`).then((response) => response.json());
    findAllFAAs = () => fetch(`${API_ROOT}/api/faas`).then((response) => response.json());
    addFAA = (newAnswer) =>
        fetch(`${API_ROOT}/api/faas/`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(newAnswer)
        }).then((response) => response.json());

    editFAA = (id, newAnswer) =>
        fetch(`${API_ROOT}/api/faas/${id}`, {
            method: 'PUT',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(newAnswer)
        }).then((response) => response.json());
}