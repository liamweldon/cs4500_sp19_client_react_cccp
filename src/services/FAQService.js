import {API_ROOT} from '../api-config';

export default class FAQService {
  static instance = null;
  static getInstance() {
    if (FAQService.instance === null) {
      FAQService.instance = new FAQService();
    }
    return this.instance;
  }

  addFAQ = (newQuestion) =>
    fetch(`${API_ROOT}/api/faqs/`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(newQuestion)
    }).then((response) => response.json());

  editFAQ = (id, newQuestion) =>
    fetch(`${API_ROOT}/api/faqs/${id}`, {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(newQuestion)
    }).then((response) => response.json());
  findFAQById = (id) => fetch(`${API_ROOT}/api/faqs/${id}`).then((response) => response.json());
  deleteFAQ = (id) => fetch(`${API_ROOT}/api/faqs/${id}`, {method: 'DELETE'}).then((response) => response.json());
  findAllFAQs = () => fetch(`${API_ROOT}/api/faqs`).then((response) => response.json());
  linkFAAtoFAQ = (faaID, faqID) => fetch(`${API_ROOT}/api/faqs/${faqID}/faas/${faaID}`).then(response => response.json());
}
