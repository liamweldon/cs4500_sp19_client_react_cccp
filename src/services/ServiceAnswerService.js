export default class ServiceAnswerService {
    static instance = null;
    static getInstance() {
        if(ServiceAnswerService.instance === null) {
            ServiceAnswerService.instance = new ServiceAnswerService()
        }
        return this.instance
    }
    findServiceAnswerById = id =>
    fetch(`http://localhost:8080/api/service-answers/${id}`)
    .then(response => response.json())
    findAllServiceAnswers = () =>
    fetch("http://localhost:8080/api/service-answers")
    .then(response => response.json())
}