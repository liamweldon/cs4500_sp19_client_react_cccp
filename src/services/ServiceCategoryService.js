export default class ServiceCategoryService {
    static instance = null;
    static getInstance() {
        if(ServiceCategoryService.instance === null) {
            ServiceCategoryService.instance = new ServiceCategoryService()
        }
        return this.instance
    }
    findServiceCategoryById = categoryId =>
        fetch(`http://localhost:8080/api/service-categories/${categoryId}`)
            .then(response => response.json())
    findAllServiceCategories = () =>
        fetch("http://localhost:8080/api/service-categories")
            .then(response => response.json())
}
