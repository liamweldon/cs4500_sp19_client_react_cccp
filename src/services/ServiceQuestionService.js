import {API_ROOT} from '../api-config';

export default class ServiceQuestionService {
    static instance = null;
    static getInstance() {
        if (ServiceQuestionService.instance === null) {
            ServiceQuestionService.instance = new ServiceQuestionService();
        }
        return this.instance;
    }
    findServiceQuestionById = (id) =>
        fetch(`${API_ROOT}/api/service-questions/${id}`).then((response) => response.json());

    findAllServiceQuestions = () => fetch(`${API_ROOT}/api/service-questions`).then((response) => response.json());

    addServiceQuestion = (newServiceQuestion) =>
    fetch(`${API_ROOT}/api/service-questions/`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(newServiceQuestion)
    }).then((response) => response.json());
}
