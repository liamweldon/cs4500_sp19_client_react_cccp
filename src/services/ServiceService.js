import { API_ROOT } from "../api-config";

export default class ServiceService {
  static instance = null;

  static getInstance() {
    if (ServiceService.instance === null) {
      ServiceService.instance = new ServiceService();
    }
    return this.instance;
  }

  findServiceById = serviceId =>
    fetch(`${API_ROOT}/api/services/${serviceId}/`).then(response =>
      response.json());

  findAllServices = () =>
    fetch(`${API_ROOT}/api/services`).then(response => response.json());

  createService = service => {
    delete service.id;
    return fetch(`${API_ROOT}/api/services`,
    {
      method: 'post',
      body: JSON.stringify(service),
      headers: {'content-type': 'application/json'}
    })
    .then(response => response.json())
  };

  updateService = service => {
    return fetch(`${API_ROOT}/api/services/${service.id}`,
    {
      method: 'put',
      body: JSON.stringify(service),
      headers: {'content-type': 'application/json'}
    })
    .then(response => response.json())
  };

  deleteService = id => {
    return fetch(`${API_ROOT}/api/services/${id}`,
    {
      method: 'delete'
    })
  };

}
