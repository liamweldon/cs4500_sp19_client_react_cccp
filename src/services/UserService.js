import {API_ROOT} from '../api-config';

export default class UserService {
    static instance = null;

    static getInstance() {
        if (UserService.instance === null) {
            UserService.instance = new UserService();
        }
        return this.instance;
    }

    findUserById = (userId) => fetch(`${API_ROOT}/api/users/${userId}`).then((response) => response.json());
    findAllUsers = () => fetch(`${API_ROOT}/api/users`).then((response) => response.json());
    // TODO update urls
    createUser = user =>
        fetch(`${API_ROOT}/api/users`,
            {
                method: 'POST',
                body: JSON.stringify(user),
                headers: {'content-type': 'application/json'}
            })
            .then(response => response.json())

    updateUser = user =>
        fetch(`${API_ROOT}/api/users/${user.id}`,
            {
                method: 'PUT',
                body: JSON.stringify(user),
                headers: {'content-type': 'application/json'}
            })
            .then(response => response.json())

    deleteUser = user =>
        fetch(`${API_ROOT}/api/users/${user.id}`,
            {
                method: 'DELETE',
            })
}
