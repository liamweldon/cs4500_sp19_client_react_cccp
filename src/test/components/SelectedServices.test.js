import React from "react";
import TestRenderer from "react-test-renderer";

import SelectedServices from "../../components/BusinessServiceScreen/SelectedServices";
import ServiceService from "../../services/ServiceService";

import services from "../mockData/services.mock.json";
import mockServiceService from "../services/mocks/MockServiceService";

const serviceService = ServiceService.getInstance();
beforeEach(() => {
  global.fetch = mockServiceService(services);
});

let newServices = services.slice(0, 1); 
let clickedService = undefined;

// Mock out selecting a service
const clickService = () => {
	clickedService =  newServices.pop()
}

test("Test clicking a service renders properly", () => {

  return serviceService
    .findAllServices()
    .then(services => {
      let component = TestRenderer.create(
        <SelectedServices selectedServices={newServices} clickedSelectedService={clickedService}
         />
      );
      let tree = component.toJSON();
      let testInstance = component.root;
      expect(tree).toMatchSnapshot();

      let rows = testInstance.findAllByProps({
            className: "list-group-item"
      });
      expect(rows.length).toBe(1);

      clickService();

      // Simulate clicking a service, removing it from the selected list.
      component = TestRenderer.create(
        <SelectedServices selectedServices={newServices} clickedSelectedServices={clickedService} />
      );
      tree = component.toJSON();
      testInstance = component.root;
      expect(tree).toMatchSnapshot();

      rows = testInstance.findAllByProps({
            className: "list-group-item"
      });
      expect(rows.length).toBe(0);
	});
})