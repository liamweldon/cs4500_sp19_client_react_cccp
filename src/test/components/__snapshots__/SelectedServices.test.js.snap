// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test clicking a service renders properly 1`] = `
<div>
  <h4>
    Selected Services
  </h4>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
      onClick={[Function]}
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <div>
        House Cleaners
        , 
        123
      </div>
      <div
        onClick={[Function]}
      >
        <i
          className="fa fa-times"
        />
      </div>
    </li>
  </ul>
</div>
`;

exports[`Test clicking a service renders properly 2`] = `
<div>
  <h4>
    Selected Services
  </h4>
  <ul
    className="list-group"
  />
</div>
`;
